lib/pomelo_ex/general/users/search_user.ex

```diff
@@ -1,7 +1,60 @@
 defmodule PomeloEx.General.Users.SearchUser do
   @moduledoc false
 
-  def execute(payload) do
-    raise "Not implemented #{__MODULE__} payload:" <> inspect(payload)
+  alias PomeloEx.General.Authorization
+  alias PomeloEx.Types.General.Users.SearchUserType
+
+  def execute(%SearchUserType{} = payload) do
+    http_client = Application.get_env(:pomelo_ex, :http_adapter)
+    url = Application.get_env(:pomelo_ex, :url)
+    headers = build_headers()
+
+    params =
+      payload
+      |> to_query_params()
+      |> URI.encode_query()
+
+    http_client.get("#{url}/users/v1/?#{params}", headers)
+  end
+
+  defp build_headers do
+    {:ok, %HTTPoison.Response{status_code: 200, body: body}} = Authorization.request_token()
```

Comment:
I would need to pass this token as a parameter to the execute function. The idea behind this token is for the client to reuse it until it expires; however, in this case, a new token is being generated with each request.